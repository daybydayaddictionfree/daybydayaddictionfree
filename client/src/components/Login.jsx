import React from 'react';
import GoogleLogin from 'react-google-login';
import GoogleId from '../../../keysAndIds/GoogleId.js';

const responseGoogle = (response) => {
  console.log(response);
};

const Login = props => (
  <GoogleLogin
    clientId={GoogleId.client_id}
    buttonText="Login using Google"
    onSuccess={props.responseGoogle}
    onFailure={props.responseGoogle}
  />
);

export default Login;
