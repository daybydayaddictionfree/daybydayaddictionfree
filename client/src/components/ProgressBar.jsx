import React from 'react';
import LinearProgress from 'material-ui/LinearProgress';

class ProgressBar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      /* this number will update from the database everytime a
      smoker has a successful or unsuccessful day */
      completed: 30,
    };
  }

  // call this or a similar function when database is updated
  progress(value) {
    this.setState({ completed: value });
  }

  render() {
    return (
      <LinearProgress style={{ width: '66%', margin: '5px auto' }} mode="determinate" max={60} value={this.state.completed} />
    );
  }
}

export default ProgressBar;
