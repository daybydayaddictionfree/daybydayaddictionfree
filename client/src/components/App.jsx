import React from 'react';
import { Link, Route } from 'react-router-dom';
import UserPage from './UserPage.jsx';
import Login from './Login.jsx';
import SignUp from './signUp.jsx';

const App = () => (
  <div>
    <nav>
      <Link to="/" style={{ margin: '5px' }} >LandingPage</Link>
      <Link to="/signup" style={{ margin: '5px' }} >SigupPage</Link>
      <Link to="/user" style={{ margin: '5px' }} >UserPage</Link>
      <Login />
    </nav>
    <div>
      <Route exact path="/" component={LandingPage} />
      <Route exact path="/signup" component={SignupPage} />
      <Route exact path="/user" component={UserPage} />
    </div>
  </div>
);

const LandingPage = () => <h1>Future Landing Page</h1>;
const SignupPage = () => <SignUp/>;

export default App;
