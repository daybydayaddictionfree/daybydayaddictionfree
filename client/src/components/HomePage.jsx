import React from 'react';
import axios from 'axios';
import MessageList from './MessageList.jsx'
import ProgressBar from './ProgressBar.jsx';
import { GoogleLogout } from 'react-google-login';

const HomePage = (props) => {
  const logout = () => {
    axios.get('/logout')
      .then(() => {
        props.cookies.remove('dbd-session-cookie');
      })
      .then(() => {
        props.logout();
      })
      .catch(err => console.log('Error logging out ', err));

  };

  return (
    <div>
      <GoogleLogout
        buttonText="Logout"
        onLogoutSuccess={logout}
      />
      <h1 style={{ textAlign: 'center' }} >{props.userState.profileObj.givenName} {props.userState.profileObj.familyName}</h1>
      <ProgressBar />
      <MessageList messages={props.messages} />
    </div>
  );
}

export default HomePage;
