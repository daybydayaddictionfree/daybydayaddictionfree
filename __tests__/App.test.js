import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import 'raf/polyfill';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/src/components/App';
import HomePage from '../client/src/components/HomePage';
import MessageList from '../client/src/components/MessageList';
import Message from '../client/src/components/Message';
import ProgressBar from '../client/src/components/ProgressBar';
import Login from '../client/src/components/Login';
import messages from './../sampleData';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('should be a stateful class component ', () => {
    const wrapper = shallow( <App /> );
    expect(React.Component.isPrototypeOf(App)).toBe(true);
  });

  it('should render the App component ', () => {
    const wrapper = shallow( <App /> );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<HomePage />', () => {
  it('should be a stateless component ', () => {
    expect(React.Component.isPrototypeOf(HomePage)).toBe(false);
  });

  it('should render the HomePage component ', () => {
    const userState = {
      profileObj: {
        givenName: 'Brian',
        familyName: 'Choe',
      }
    };
    const wrapper = shallow( <HomePage userState={userState} /> );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<MessageList />', () => {
  it('should be a stateless component ', () => {
    expect(React.Component.isPrototypeOf(MessageList)).toBe(false);
  });

  it('should render the MessageList component ', () => {
    const wrapper = shallow( <MessageList messages={messages} /> );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Message />', () => {
  it('should be a stateless component ', () => {
    expect(React.Component.isPrototypeOf(Message)).toBe(false);
  });

  it('should render the Message component ', () => {
    const wrapper = shallow( <Message message={messages[0]} /> );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<ProgressBar />', () => {
  it('should be a stateless component ', () => {
    expect(React.Component.isPrototypeOf(ProgressBar)).toBe(false);
  });

  it('should render the ProgressBar component ', () => {
    const wrapper = shallow( <ProgressBar /> );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Login />', () => {
  it('should be a stateless component ', () => {
    expect(React.Component.isPrototypeOf(Login)).toBe(false);
  });

  it('should render the Login component ', () => {
    const wrapper = shallow( <Login /> );
    expect(wrapper).toMatchSnapshot();
  });
});
