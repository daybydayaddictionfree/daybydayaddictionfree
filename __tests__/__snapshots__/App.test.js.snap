// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render the App component  1`] = `
<div>
  <div
    className="btn-right"
  >
    <nav>
      <Login
        buttonText="Login Using Google"
        responseGoogle={[Function]}
      />
    </nav>
  </div>
  <div>
    <Switch>
      <Route
        exact={true}
        path="/landing"
        render={[Function]}
      />
      <Redirect
        push={false}
        to="/landing"
      />
    </Switch>
  </div>
</div>
`;

exports[`<HomePage /> should render the HomePage component  1`] = `
<div>
  <div
    className="top-row"
  >
    <div
      className="btn-right"
    >
      <t
        buttonText="Logout"
        disabledStyle={
          Object {
            "opacity": 0.6,
          }
        }
        onLogoutSuccess={[Function]}
        onRequest={[Function]}
        responseType="permission"
        tag="button"
      />
    </div>
  </div>
  <div>
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
       
    </h1>
    <ProgressBar />
    <MessageList />
  </div>
</div>
`;

exports[`<Login /> should render the Login component  1`] = `
<t
  buttonText="Login with Google"
  clientId="661794075574-q87lqva7vp50d04qv7h4dm95lhmvs3m3.apps.googleusercontent.com"
  cookiePolicy="single_host_origin"
  disabledStyle={
    Object {
      "opacity": 0.6,
    }
  }
  fetchBasicProfile={true}
  isSignedIn={false}
  onFailure={[Function]}
  onRequest={[Function]}
  prompt=""
  scope="profile email"
  tag="button"
  type="button"
  uxMode="popup"
/>
`;

exports[`<Message /> should render the Message component  1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
  style={
    Object {
      "margin": "5px auto",
      "width": "66%",
    }
  }
>
  <CardHeader
    avatar={null}
  />
  <CardText>
    great work, keep it up!
  </CardText>
</Card>
`;

exports[`<MessageList /> should render the MessageList component  1`] = `
<div>
  <Message
    message={
      Object {
        "_id": 1,
        "sender": "Abraham Lincoln",
        "text": "great work, keep it up!",
        "timeStamp": "datestring here",
      }
    }
  />
  <Message
    message={
      Object {
        "_id": 2,
        "sender": "George Washington",
        "text": "amazing!",
        "timeStamp": "next datestring here",
      }
    }
  />
  <Message
    message={
      Object {
        "_id": 3,
        "sender": "Barack Obama",
        "text": "I believe in you",
        "timeStamp": "3rd datestring here",
      }
    }
  />
  <Message
    message={
      Object {
        "_id": 4,
        "sender": "Bill Clinton",
        "text": "you can do it!",
        "timeStamp": "datestring #4 here",
      }
    }
  />
</div>
`;

exports[`<ProgressBar /> should render the ProgressBar component  1`] = `
<div>
  <div>
    <p
      className="progress-comment"
    >
      Your smoke free life begins now!
    </p>
  </div>
  <div
    className="progress-container"
  >
    <div
      className="axis"
    >
      0
    </div>
    <LinearProgress
      max={60}
      min={0}
      mode="determinate"
      style={
        Object {
          "height": "10px",
          "margin": "30px 5px",
          "width": "50%",
        }
      }
      value={0}
    />
    <div
      className="axis"
    >
      60
    </div>
  </div>
</div>
`;
